sdk/core/azure-core/test/ut/operation_status.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <gtest/gtest.h>
+
+#include <azure/core/operation_status.hpp>
+
+using namespace Azure::Core;
+
+TEST(OperationStatus, Basic)
+{
+  OperationStatus status = OperationStatus::Cancelled;
+  EXPECT_TRUE(status == OperationStatus::Cancelled);
+  EXPECT_TRUE(status.Get() == "Cancelled");
+  EXPECT_FALSE(status != OperationStatus::Cancelled);
+
+  status = OperationStatus::Failed;
+  EXPECT_TRUE(status == OperationStatus::Failed);
+  EXPECT_TRUE(status.Get() == "Failed");
+  EXPECT_FALSE(status != OperationStatus::Failed);
+
+  status = OperationStatus::NotStarted;
+  EXPECT_TRUE(status == OperationStatus::NotStarted);
+  EXPECT_TRUE(status.Get() == "NotStarted");
+  EXPECT_FALSE(status != OperationStatus::NotStarted);
+
+  status = OperationStatus::Running;
+  EXPECT_TRUE(status == OperationStatus::Running);
+  EXPECT_TRUE(status.Get() == "Running");
+  EXPECT_FALSE(status != OperationStatus::Running);
+
+  status = OperationStatus::Succeeded;
+  EXPECT_TRUE(status == OperationStatus::Succeeded);
+  EXPECT_TRUE(status.Get() == "Succeeded");
+  EXPECT_FALSE(status != OperationStatus::Succeeded);
+}
+
+TEST(OperationStatus, Custom)
+{
+  OperationStatus status2("CustomValue");
+  EXPECT_TRUE(status2.Get() == "CustomValue");
+  EXPECT_TRUE(status2 != OperationStatus::NotStarted);
+
+  OperationStatus status3 = OperationStatus("CustomValue");
+  EXPECT_TRUE(status3.Get() == "CustomValue");
+  EXPECT_TRUE(status3 != OperationStatus::NotStarted);
```

Comment:
Two different `OperationStatus` instances with the same string value are considered equal, correct? Add a test assert:

```suggestion
  EXPECT_TRUE(status3 != OperationStatus::NotStarted);
  EXPECT_EQ(status3, status2);
```

Reply:
No they won't be equal and hence the check.  status3 is a custom value and hence it should not be equal to some other status value.

Reply:
Hmm... status2 and status3 had the exact same `"CustomValue"` string though.